test/benchmark/common/host_guest_bench_runner.sh

```diff
@@ -29,26 +29,39 @@ fi
 # Parameters:
 #   $1: guest command to run on the VM
 #   $2: output file to store the benchmark results
-#   $3: sleep time before running the host command
+#   $3: log file to monitor for the start signal
+#   $4: start signal string
 run_benchmark() {
     local guest_cmd=$1
     local output_file=$2
-    local sleep_time=$3
+    local log_file=$3
+    local start_signal=$4
 
     echo "Running the benchmark on the VM..."
-    eval "${guest_cmd}" & 
-    sleep "${sleep_time}"  
+    eval "${guest_cmd}" | tee "${log_file}" & 
+
+    # Monitor the log file for the start signal
+    echo "Waiting for the start signal: ${start_signal}"
+    while true; do
+        if grep -q "${start_signal}" "${log_file}"; then
+            echo "Start signal detected. Running the benchmark on the host..."
+            break
+        fi
+        sleep 5
+    done
+
```

Comment:
Insert a `sleep 1` command here. Because even if when you see the "start signal", the guest may not be ready.

```
echo "All preparations are done. Starting the benchmark..."
echo "Running iperf3 server..."
/benchmark/bin/iperf3 -s -B 10.0.2.15 --one-off
```

In the TCP benchmark, after writing the "start signal", it still takes a small amount of time to start the iperf server. So to be more robust, the host side needs to sleep. Even with this extra sleep, the new approach based on reading a log file is still better than the current approach.